src/core/server/config/raw_config_service.ts

```diff
@@ -46,12 +40,12 @@ export class RawConfigService {
    * As we have a notion of a _current_ config we rely on a BehaviorSubject so
    * every new subscription will immediately receive the current config.
    */
-  private readonly rawConfigFromFile$: BehaviorSubject<any> = new BehaviorSubject(notRead);
+  private readonly rawConfigFromFile$: BehaviorSubject<symbol> = new BehaviorSubject(notRead);
```

Comment:
__issue:__ The very first value will be `symbol`, but consequent values will be `config`'s.

Reply:
I've tried a couple of things to type this differently, but i think the problem is `any` does not cover the `Symbol` type. 

here's one of the things i tried, but there's an error with type when we call next on the behavior subject.

```js
import { isEqual, isPlainObject } from 'lodash';
import { BehaviorSubject, from, Observable } from 'rxjs';
import { filter, map, skipWhile } from 'rxjs/operators';
import typeDetect from 'type-detect';

import { ObjectToRawConfigAdapter } from './object_to_raw_config_adapter';
import { RawConfig } from './raw_config';
import { getConfigFromFile } from './read_config';

// Used to indicate that no config has been received yet
const notRead: symbol = Symbol('config not yet read');

export class RawConfigService {
  /**
   * The stream of configs read from the config file. Will be the symbol
   * `notRead` before the config is initially read, and after that it can
   * potentially be `null` for an empty yaml file.
   *
   * This is the _raw_ config before any overrides are applied.
   *
   * As we have a notion of a _current_ config we rely on a BehaviorSubject so
   * every new subscription will immediately receive the current config.
   */
  private readonly rawConfigFromFile$:
    | BehaviorSubject<symbol>
    | BehaviorSubject<any> = new BehaviorSubject(notRead);

  private readonly config$: Observable<RawConfig>;

  constructor(readonly configFile: string) {
    this.config$ = from(this.rawConfigFromFile$).pipe(
      filter(rawConfig => rawConfig !== notRead),
      map(rawConfig => {
        // If the raw config is null, e.g. if empty config file, we default to
        // an empty config
        if (rawConfig == null) {
          return new ObjectToRawConfigAdapter({});
        }

        if (isPlainObject(rawConfig)) {
          // TODO Make config consistent, e.g. handle dots in keys
          return new ObjectToRawConfigAdapter(rawConfig);
        }

        throw new Error(`the raw config must be an object, got [${typeDetect(rawConfig)}]`);
      }),
      // We only want to update the config if there are changes to it
      skipWhile(isEqual)
    );
  }

  /**
   * Read the initial Kibana config.
   */
  public loadConfig() {
    const config = getConfigFromFile(this.configFile);
    this.rawConfigFromFile$.next(config); // error on this line
  }

  public stop() {
    this.rawConfigFromFile$.complete();
  }

  /**
   * Re-read the Kibana config.
   */
  public reloadConfig() {
    this.loadConfig();
  }

  public getConfig$() {
    return this.config$;
  }
}
```

Reply:
Here's another way, remove the symbol and behavior subject type annotations and there's a similar error on the same line

```js
import { isEqual, isPlainObject } from 'lodash';
import { BehaviorSubject, from, Observable } from 'rxjs';
import { filter, map, skipWhile } from 'rxjs/operators';
import typeDetect from 'type-detect';

import { ObjectToRawConfigAdapter } from './object_to_raw_config_adapter';
import { RawConfig } from './raw_config';
import { getConfigFromFile } from './read_config';

// Used to indicate that no config has been received yet
const notRead = Symbol('config not yet read');

export class RawConfigService {
  /**
   * The stream of configs read from the config file. Will be the symbol
   * `notRead` before the config is initially read, and after that it can
   * potentially be `null` for an empty yaml file.
   *
   * This is the _raw_ config before any overrides are applied.
   *
   * As we have a notion of a _current_ config we rely on a BehaviorSubject so
   * every new subscription will immediately receive the current config.
   */
  private readonly rawConfigFromFile$ = new BehaviorSubject(notRead);

  private readonly config$: Observable<RawConfig>;

  constructor(readonly configFile: string) {
    this.config$ = from(this.rawConfigFromFile$).pipe(
      filter(rawConfig => rawConfig !== notRead),
      map(rawConfig => {
        // If the raw config is null, e.g. if empty config file, we default to
        // an empty config
        if (rawConfig == null) {
          return new ObjectToRawConfigAdapter({});
        }

        if (isPlainObject(rawConfig)) {
          // TODO Make config consistent, e.g. handle dots in keys
          return new ObjectToRawConfigAdapter(rawConfig);
        }

        throw new Error(`the raw config must be an object, got [${typeDetect(rawConfig)}]`);
      }),
      // We only want to update the config if there are changes to it
      skipWhile(isEqual)
    );
  }

  /**
   * Read the initial Kibana config.
   */
  public loadConfig() {
    const config = getConfigFromFile(this.configFile);
    this.rawConfigFromFile$.next(config); // Argument of type 'any' is not assignable to parameter of type 'unique symbol'.
  }

  public stop() {
    this.rawConfigFromFile$.complete();
  }

  /**
   * Re-read the Kibana config.
   */
  public reloadConfig() {
    this.loadConfig();
  }

  public getConfig$() {
    return this.config$;
  }
}
```

Reply:
Can you just do something like this instead?
```ts
const notRead = Symbol('config not yet read');
.....
private readonly rawConfigFromFile$ = new BehaviorSubject<any>(notRead);
```
or
```ts 
const notRead = Symbol('config not yet read');
.....
private readonly rawConfigFromFile$ = new BehaviorSubject<symbol | Record<string, any>>(
  notRead
);
.....
map((rawConfig: Record<string, any>) => {
```

Reply:
The former works! I wonder why compiler doesn't complain about `unique symbol` not being `any` type when done this way.

Reply:
@epixa Here's the weirdness I was talking about.